### PR TITLE
chipsec: 1.5.1 -> 1.6.1

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 1.5.10

Change log: https://github.com/chipsec/chipsec/releases/tag/1.5.10

This is the migration to Python3. 1.5.2 was the last release with support for Python 2.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 117624` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.chipsec</li>
    <li>linuxPackages_latest-libre.chipsec</li>
  </ul>
</details>
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>chipsec</li>
    <li>linuxPackages.chipsec</li>
    <li>linuxPackages_4_14.chipsec</li>
    <li>linuxPackages_4_19.chipsec</li>
    <li>linuxPackages_4_4.chipsec</li>
    <li>linuxPackages_4_9.chipsec</li>
    <li>linuxPackages_5_10.chipsec</li>
    <li>linuxPackages_5_11.chipsec</li>
    <li>linuxPackages_hardened.chipsec</li>
    <li>linuxPackages_latest_hardened.chipsec</li>
    <li>linuxPackages_latest_xen_dom0.chipsec</li>
    <li>linuxPackages_latest_xen_dom0_hardened.chipsec</li>
    <li>linuxPackages_lqx.chipsec</li>
    <li>linuxPackages_testing_bcachefs.chipsec</li>
    <li>linuxPackages_xen_dom0.chipsec</li>
    <li>linuxPackages_xen_dom0_hardened.chipsec</li>
    <li>linuxPackages_zen.chipsec</li>
  </ul>
</details>

**Comment by JohnAZoidberg** [comment]
> This is the migration to Python3. 1.5.2 was the last release with support for Python 2.

Awesome, finally!

How did you test your changes?

**Review by JohnAZoidberg — COMMENTED** [review]
Hmm, doesn't work for me :/

```
> sudo ./result/bin/chipsec_main 
################################################################
##                                                            ##
##  CHIPSEC: Platform Hardware Security Assessment Framework  ##
##                                                            ##
################################################################
[CHIPSEC] Version 1.5.10
[CHIPSEC] Arguments: 

ERROR: Message: "Cannot find chipsec.ko module"
> sudo ./result/bin/chipsec_util uefi var-list

################################################################
##                                                            ##
##  CHIPSEC: Platform Hardware Security Assessment Framework  ##
##                                                            ##
################################################################
[CHIPSEC] Version : 1.5.10
[CHIPSEC] OS      : Linux 5.11.8 #1-NixOS SMP Sat Mar 20 09:51:14 UTC 2021 x86_64
[CHIPSEC] Python  : 3.8.8 (64-bit)

ERROR: Message: "Cannot find chipsec.ko module"
```

File: pkgs/tools/security/chipsec/default.nix [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
+, stdenv
+, fetchFromGitHub
+, kernel ? null
+, libelf
+, nasm
+, python3
+, withDriver ? false
+}:
+
+python3.pkgs.buildPythonApplication rec {
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/tools/security/chipsec/default.nix:11]: python3Packages is an alias to python3.pkgs.

[the remark this answers is not in the document]
Comment (JohnAZoidberg) [reply pkgs/tools/security/chipsec/default.nix:11]: Oh I see, and it's quite commonly used as well.
https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/NixOS/nixpkgs%24+python3.pkgs&patternType=literal

**Comment by JohnAZoidberg** [comment]
[1.6.0](https://github.com/chipsec/chipsec/releases/tag/1.6.0) was just released.

**Comment by JohnAZoidberg** [comment]
And [1.6.1](https://github.com/chipsec/chipsec/releases/tag/1.6.1)

**Comment by JohnAZoidberg** [comment]
Ok finally I was able to build it and run it with the kernel module.

```
$ nix-build -A linuxPackages_5_11.chipsec
[... skipped ...]

$ sudo ./result/bin/chipsec_main
################################################################
##                                                            ##
##  CHIPSEC: Platform Hardware Security Assessment Framework  ##
##                                                            ##
################################################################
[CHIPSEC] Version 1.6.1
[CHIPSEC] Arguments:

****** Chipsec Linux Kernel module is licensed under GPL 2.0
[CHIPSEC] API mode: using CHIPSEC kernel module API
[CHIPSEC] OS      : Linux 5.11.8 #1-NixOS SMP Sat Mar 20 09:51:14 UTC 2021 x86_64
[CHIPSEC] Python  : 3.8.8 (64-bit)
[CHIPSEC] Helper  : LinuxHelper (/nix/store/fzfxhcasr1hwx78s8gjfl1dwgf38pwv3-chipsec-1.6.1/lib/python3.8/site-packages/drivers/linux/chipsec.ko)
[CHIPSEC] Platform: Mobile 8th Generation Core Processor (Whiskey Lake U 4 Cores)
[CHIPSEC]      VID: 8086
[CHIPSEC]      DID: 3E34
[CHIPSEC]      RID: 0C
[CHIPSEC] PCH     : Intel 300 series On-Package PCH
[CHIPSEC]      VID: 8086
[CHIPSEC]      DID: 9D84
[CHIPSEC]      RID: 11

[... skipped ...]

[CHIPSEC] ***************************  SUMMARY  ***************************
[CHIPSEC] Time elapsed            0.629
[CHIPSEC] Modules total           27
[CHIPSEC] Modules failed to run   0:
[CHIPSEC] Modules passed          21:
[+] PASSED: chipsec.modules.common.bios_kbrd_buffer
[+] PASSED: chipsec.modules.common.bios_smi
[+] PASSED: chipsec.modules.common.bios_ts
[+] PASSED: chipsec.modules.common.bios_wp
[+] PASSED: chipsec.modules.common.cpu.spectre_v2
[+] PASSED: chipsec.modules.common.debugenabled
[+] PASSED: chipsec.modules.common.ia32cfg
[+] PASSED: chipsec.modules.common.me_mfg_mode
[+] PASSED: chipsec.modules.common.memconfig
[+] PASSED: chipsec.modules.common.memlock
[+] PASSED: chipsec.modules.common.remap
[+] PASSED: chipsec.modules.common.secureboot.variables
[+] PASSED: chipsec.modules.common.smm
[+] PASSED: chipsec.modules.common.smm_code_chk
[+] PASSED: chipsec.modules.common.smm_dma
[+] PASSED: chipsec.modules.common.smrr
[+] PASSED: chipsec.modules.common.spd_wd
[+] PASSED: chipsec.modules.common.spi_desc
[+] PASSED: chipsec.modules.common.spi_fdopss
[+] PASSED: chipsec.modules.common.spi_lock
[+] PASSED: chipsec.modules.common.uefi.access_uefispec
[CHIPSEC] Modules information     1:
[#] INFORMATION: chipsec.modules.common.cpu.cpu_info
[CHIPSEC] Modules failed          0:
[CHIPSEC] Modules with warnings   4:
[!] WARNING: chipsec.modules.common.rtclock
[!] WARNING: chipsec.modules.common.sgx_check
[!] WARNING: chipsec.modules.common.spi_access
[!] WARNING: chipsec.modules.common.uefi.s3bootscript
[CHIPSEC] Modules not implemented 0:
[CHIPSEC] Modules not applicable  1:
[*] NOT APPLICABLE: chipsec.modules.common.cpu.ia_untrusted
[CHIPSEC] *****************************************************************
```

**Review by JohnAZoidberg — APPROVED** [review]

**Review by JohnAZoidberg — CHANGES_REQUESTED** [review]
~~Ahh wait, I broke the build without kernel modules~~

**Comment by fabaff** [comment]
@JohnAZoidberg, thanks for your support.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 117624` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.chipsec</li>
    <li>linuxPackages_hardkernel_4_14.chipsec</li>
    <li>linuxPackages_hardkernel_latest.chipsec</li>
    <li>linuxPackages_latest-libre.chipsec</li>
  </ul>
</details>
<details>
  <summary>18 packages built:</summary>
  <ul>
    <li>chipsec</li>
    <li>linuxPackages.chipsec (linuxPackages_5_10.chipsec)</li>
    <li>linuxPackages_4_14.chipsec</li>
    <li>linuxPackages_4_19.chipsec</li>
    <li>linuxPackages_4_4.chipsec</li>
    <li>linuxPackages_4_9.chipsec</li>
    <li>linuxPackages_5_11.chipsec (linuxPackages_latest.chipsec)</li>
    <li>linuxPackages_5_4.chipsec</li>
    <li>linuxPackages_hardened.chipsec</li>
    <li>linuxPackages_latest_hardened.chipsec</li>
    <li>linuxPackages_latest_xen_dom0.chipsec</li>
    <li>linuxPackages_latest_xen_dom0_hardened.chipsec</li>
    <li>linuxPackages_lqx.chipsec</li>
    <li>linuxPackages_testing_bcachefs.chipsec</li>
    <li>linuxPackages_xanmod.chipsec</li>
    <li>linuxPackages_xen_dom0.chipsec</li>
    <li>linuxPackages_xen_dom0_hardened.chipsec</li>
    <li>linuxPackages_zen.chipsec</li>
  </ul>
</details>

**Comment by JohnAZoidberg** [comment]
Hmm, not sure why ofBorg is having an issue.

**Pull request merged**